tests/test_internals/test_si.py

```diff
@@ -576,5 +585,173 @@ def test_group_set_custom_description(self):
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(rel_x=0.0, rel_y=0.0, id=124, channel_name=str(124))
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(rel_x=0.0, rel_y=0.0, id=None, channel_name="6")  # automatic ID set
+        self.nwbfile.add_electrode(**values_dic)
+
+        expected_ids = [0, 1, 2, 3, 123, 124, 6]
+        expected_names = ["a", "b", "c", "d", "123", "124", "6"]
+        self.assertListEqual(list(self.nwbfile.electrodes.id.data), expected_ids)
+        self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), expected_names)
+
+    def test_manual_row_adition_before_add_electrodes_function_and_channel_name_collision(self):
+        """
+        Add some rows to the electrode tables before using the add_electrodes function.
+        In this case there is with some common channel names between the previously added rows and the recroder
+        which causes collisions.
+        """
+
+        values_dic = self.defaults
+        self.nwbfile.add_electrode_column(name="channel_name", description="a string reference for the channel")
+        self.nwbfile.add_electrode_column(name="property1", description="property_added_before")
+
+        values_dic.update(id=20, channel_name="c", property1="value1_c")
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(id=21, channel_name="d", property1="value1_d")
+        self.nwbfile.add_electrode(**values_dic)
+
+        property_values = ["value2_a", "value2_b", "value2_c", "value2_d"]
+        self.recording_1.set_property(key="property2", values=property_values)
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+
+        expected_ids = [20, 21, 2, 3]
+        expected_names = ["c", "d", "a", "b"]
+        self.assertListEqual(list(self.nwbfile.electrodes.id.data), expected_ids)
+        self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), expected_names)
+
+        expected_property_values1 = ["value1_c", "value1_d", "", ""]
+        expected_property_values2 = ["value2_c", "value2_d", "value2_a", "value2_b"]
+        self.assertListEqual(list(self.nwbfile.electrodes["property1"].data), expected_property_values1)
+        self.assertListEqual(list(self.nwbfile.electrodes["property2"].data), expected_property_values2)
+
+    def test_manually_added_before_recording_id_collision(self):
+        """
+        Add some rows to the electrode tables before using the add_electrodes function.
+        In this case there is with some common ids between the manually added rows and the recorder which causes collisions.
+        """
+
+        values_dic = self.defaults
+
+        values_dic.update(id=0)
+        self.nwbfile.add_electrode(**values_dic)
+
+        values_dic.update(id=1)
+        self.nwbfile.add_electrode(**values_dic)
+
+        self.assertRaises(ValueError, add_electrodes, recording=self.base_recording, nwbfile=self.nwbfile)
```

Comment:
```suggestion
        self.assertRaisesWith(ValueError, "insert the exact error message here"):
            recording=self.base_recording, nwbfile=self.nwbfile)
```

Minor suggestion for enhanced specificity. Also always prefer to pass the callables after the context to make the usage more apparent.

Reply:
Note that `self.assertRaisesWith` is only present in the `from hdmf.testing import TestCase` library.

Reply:
What about:
```python
        self.assertRaises(ValueError, lambda: add_electrodes(recording=self.base_recording, nwbfile=self.nwbfile))
```
It is expressive without requiring tools outside of unittest. I don't have a specific message designed here, but if you want that it can be done with `assertRaisesRegex`. If you do want that, do you have any suggestions for the message? : O

Reply:
The `hdmf` `TestCase` is a child of `unittest`, and we use it elsewhere in the testing suite so you don't have to worry about that.

Will the text of the message itself change for this specific test function? If not you can simply run it once, grab the message, and fill in the text value. This adds specificity that it isn't some other `ValueError` causing a problem (fairly common error type, but not as common as `AssertionError`.)